packages/router/src/directives/router_link_active.ts

```diff
@@ -161,8 +170,13 @@ export class RouterLinkActive implements OnChanges, OnDestroy, AfterContentInit
   }
 
   private isLinkActive(router: Router): (link: (RouterLink|RouterLinkWithHref)) => boolean {
-    return (link: RouterLink|RouterLinkWithHref) =>
-               router.isActive(link.urlTree, this.routerLinkActiveOptions.exact);
+    if (isExactOptions(this.routerLinkActiveOptions)) {
+      const options = this.routerLinkActiveOptions.exact;
+      return (link: RouterLink|RouterLinkWithHref) => router.isActive(link.urlTree, options);
+    } else {
+      const options = this.routerLinkActiveOptions;
+      return (link: RouterLink|RouterLinkWithHref) => router.isActive(link.urlTree, options);
+    }
```

Comment:
It looks like you can simplify that a bit:
```suggestion
    const routerLinkActiveOptions = this.routerLinkActiveOptions;  // Internal note: this should help avoid extra payload, `routerLinkActiveOptions` would be minified.
    const options = isExactOptions(routerLinkActiveOptions)
      ? routerLinkActiveOptions.exact : routerLinkActiveOptions;
    return (link: RouterLink|RouterLinkWithHref) => router.isActive(link.urlTree, options);
```

Reply:
This is how it was before, but doesn't work now that `router.isActive` has the override.  The error is 
```
const options: boolean | IsActiveUrlTreeOptions
No overload matches this call.
  Overload 1 of 2, '(url: string | UrlTree, exact: boolean): boolean', gave the following error.
    Argument of type 'boolean | IsActiveUrlTreeOptions' is not assignable to parameter of type 'boolean'.
      Type 'IsActiveUrlTreeOptions' is not assignable to type 'boolean'.
  Overload 2 of 2, '(url: string | UrlTree, matchOptions: IsActiveUrlTreeOptions): boolean', gave the following error.
    Argument of type 'boolean | IsActiveUrlTreeOptions' is not assignable to parameter of type 'IsActiveUrlTreeOptions'.
      Type 'boolean' is not assignable to type 'IsActiveUrlTreeOptions'.ts(2769)
```

Reply:
Brilliant, thanks Pete!